cli/internal/typer/generator/core/file_manager.go

```diff
@@ -47,21 +47,21 @@ func (fm *FileManager) WriteFile(file File) error {
 }
 
 // WriteFiles writes multiple files atomically as a batch operation
-func (fm *FileManager) WriteFiles(files []File) error {
+func (fm *FileManager) WriteFiles(files []*File) error {
```

Comment:
If we are changing this interface to pointers, let's make sure the other functions also have a pointer interface (e.g WriteFile or validate)